packages/api/src/controllers/sensorController.js

```diff
@@ -188,40 +187,45 @@ const sensorController = {
         const data = await baseController.getByFieldId(sensorModel, 'farm_id', farm_id);
         res.status(200).send(data);
       } catch (error) {
-        //handle exceptions
         res.status(400).json({
           error,
         });
       }
     };
   },
 
-  // TODO
   addReading() {
     return async (req, res) => {
       // const trx = await transaction.start(Model.knex());
       try {
-        // const infoBody = {
-        //   reading_id: req.body.reading_id,
-        //   read_time: req.body.read_time,
-        //   //   transmit_time: req.body.transmit_time,
-        //   sensor_id: req.body.sensor_id,
-        //   reading_type: req.body.reading_type,
-        //   value: req.body.value,
-        //   unit: req.body.unit,
-        // };
+        const infoBody = [];
+        for (const sensor of req.body) {
+          const corresponding_sensor = await sensorModel
+            .query()
+            .select('sensor_id')
+            .where('external_id', sensor.sensor_esid)
+            .where('partner_id', req.params.partner_id);
+          for (let i = 0; i < sensor.value.length; i++) {
+            const row = {
+              read_time: sensor.time[i],
+              sensor_id: corresponding_sensor[0].sensor_id,
+              reading_type: sensor.parameter_number,
+              value: sensor.value[i],
+              unit: sensor.unit,
+            };
+            // Only include this entry if all required values are poulated
+            if (Object.values(row).every((value) => value)) {
+              infoBody.push(row);
+            }
+          }
+        }
 
-        // if (!Object.values(infoBody).every((value) => value)) {
-        //   res.status(400).send('Invalid reading');
-        // }
-        // const result = await baseController.postWithResponse(sensorReadingModel, infoBody, req, {
-        //   trx,
-        // });
-        // await trx.commit();
-        // res.status(200).send(result);
-        res.sendStatus(200);
+        const result = await baseController.postWithResponse(sensorReadingModel, infoBody, req, {
+          trx,
+        });
+        await trx.commit();
```

Comment:
You'll need to declare `trx` somewhere earlier, because I am currently getting errors that it is undefined